### PR TITLE
US-2.6 Design: Table Diff Highlighting

### PR DESCRIPTION
**Pull request by galexy**
## Summary

Design documents for US-2.6: Table Diff Highlighting — injecting highlight markup into original table HTML at `CellDiff.sourceMapping` offsets.

- **Implementation design** (`.specs/us-2-6-table-diff/implementation-design.md`): approach, algorithm, files to modify, edge cases
- **Test plan** (`.specs/us-2-6-table-diff/test-plan.md`): BDD acceptance criteria, unit/integration test scenarios, fixtures

### Key Design Decisions

1. **CSS class injection, not `<ins>`/`<del>` wrappers** — HTML table content model forbids `<ins>`/`<del>` as children of `<table>`/`<tr>`. Instead: inject CSS classes into `<tr>`/`<td>`/`<th>` opening tags.
2. **Row-level highlighting** — `<tr class="diff-row-added">` / `diff-row-removed` for whole-row changes
3. **Cell-level highlighting** — `<td class="diff-cell-modified">` with `<del>old</del> → <ins>new</ins>` annotation
4. **Single replacement pass** — table and paragraph replacements collected together, sorted reverse, applied once
5. **Backward-compatible** — `tableIndex` parameter added as optional; existing US-2.5 callers unaffected

### Beads Issue

`edgar-diff-6wf` — US-2.6: Table Diff Highlighting

## Test plan

- [ ] Review implementation design for approach and edge case coverage
- [ ] Review test plan for BDD scenario completeness
- [ ] Verify alignment between design and test plan

Co-Authored-By: SageOx <ox@sageox.ai>

### CODE REVIEW

File: .specs/us-2-6-table-diff/test-plan.md [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+  And the TableDiff has a CellDiff with changeType 'modified'
+  And the CellDiff has oldValue="$1,000" and newValue="$1,234"
+  When the filing content is rendered
+  Then the <td> element has class "diff-cell-modified"
+  And the cell inner content contains <del class="diff-removed">$1,000</del>
+  And the cell inner content contains <ins class="diff-added">$1,234</ins>
+  And a "→" separator appears between the old and new values
+```
+
+### AC-2: Added cell highlighted green
```
Comment (galexy) [inline .specs/us-2-6-table-diff/test-plan.md]: how do you add just a cell? is this really a column being added?
